### PR TITLE
Add asv update script and version benchmarks

### LINKED ISSUE #1165: Document running airspeed velocity benchmarks and set up site
#1146 and #1138 added some initial benchmarks using [airspeed velocity](https://asv.readthedocs.io/en/stable/). For the record, here's what the current set look like 

image

We have some basic benchmarks for Hudson and DTWF simulations. Getting these up and running and working for the problems I was working on was pretty frustrating and there's a lot of gotchas involved in running it. Hopefully these will settle down a bit now, though. One of the reasons it's so awkward is that ASV is built to accurately time short-lived functions, but we're mainly interested in things that take 10s of seconds (at least) to run.

Ideally, we'd like to run these regularly and keep a site updated with the results. The simplest way to do this is probably for us in Oxford to run the benchmarks on one of our servers on a cron job, and publish the results to github pages. We can look into having a dedicated machine at some point, but this will get us most of the way without extra faff of trying to find appropriate hardware.

@benjeffery, any thoughts here?

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #1165 
Site is live at https://tskit-dev.github.io/msprime-asv and being updated by a cronjob from an old laptop `poplar` in my loft.
It will slowly cover the old commits over a few days.
Specs:
 - Core i5-3317U @ 2.60GHz
 -  4GB RAM
 - Python 3.8.

We somehow ended up with two benchmarking sections in the docs - I've fixed that too.

### CODE REVIEW

File: benchmarks/benchmarks.py [hunk 1]
```diff
@@ -47,6 +47,9 @@ class LargeSimulationBenchmark:
     rounds = 1
     min_run_count = 1
     timeout = 120
+    # As we may wish to change code but not have ASV reset benchmarks we set a
+    # fixed version.
+    version = "1"
```
Comment (jeromekelleher) [inline benchmarks/benchmarks.py:52]: I wasn't sure about this - does this mean we have a single version for all the benchmarks in this class? So, if we change one do we then invalidate all the benchmarks? If we do tweak the code, isn't it most likely to be to change parameters? I think we'd be better off using ASV's approach until we're sure that it doesn't work and resort to manual versioning of each individual benchmark as a last resort.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, thanks. See comment about versioning.

File: benchmarks/benchmarks.py (hunk 1 above)
Comment (benjeffery) [reply benchmarks/benchmarks.py]: I added this in before I realised it was no bother to regenerate past benchmarks, I've removed it now.

**Comment by benjeffery** [comment]
@jeromekelleher should be good to go!

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**